software/generic/ControlEffect.cpp

```diff
@@ -0,0 +1,21 @@
+#include "ControlEffect.hpp"
+
+static inline uint8_t min(uint8_t a, uint8_t b) {
+  if (a < b) {
+    return a;
+  } else {
+    return b;
+  }
+}
+
+ControlEffect::ControlEffect(uint8_t numLeds, DeviceType deviceType)
+    : Effect(numLeds, deviceType) {}
+
+CRGB ControlEffect::GetRGB(uint8_t ledIndex, uint32_t timeMs,
+                              RadioPacket *setEffectPacket) {
+  uint8_t r = setEffectPacket->readRFromSetRgb();
```

Comment:
Looks like these functions on the packet are undefined.

Reply:
Yeah, I needed to add a bunch more to this review. Inbound